docs/markdown/Using Pants/concepts/targets.md

```diff
@@ -45,6 +45,22 @@ All target types have a `name` field, which is used to identify the target. Targ
 
 You can autoformat `BUILD` files by enabling a `BUILD` file formatter by adding it to `[GLOBAL].backend_packages` in `pants.toml` (such as `pants.backend.build_files.fmt.black` [or others](doc:enabling-backends)). Then to format, run `./pants fmt '**/BUILD'` or `./pants fmt ::` (formats everything).
 
+Environment variables
+---------------------
+
+BUILD files are very hermetic in nature with no support for using `import` or other I/O operations. In order to have dynamic data in BUILD files, you may inject values from the local environment using the `env()` function. It takes the variable name and optional default value as arguments.
```

Comment:
Sorry to snipe this review after merge. Can we document whether sensitive env vars are safe to load? Why/why not?

Reply:
Sounds like a good idea. I don't know the answer though.. :)

Reply:
@stuhood ?

Reply:
I mean... it just depends where they are being used, I suppose? Technically you could use this to interpolate an `env` var into a target name or something, at which point your secret would leak via `./pants list`.

So it's very use-site specific.